### ISSUE TITLE
🐛 Bug Report: Server Error (500) when searching (Query.equal("x", 0)) for 0 in an interger field

### ISSUE THREAD

**Issue by zwe1ch** [appwrite/appwrite#3926]
### 👟 Reproduction steps

When I search for 0 in an integer field, I get a 500 error. If I use a different integer value, the query works as expected. If I define the field as a string, the query also works. The call is made via the node-appwrite SDK with an API key and full permissions.

![grafik](https://user-images.githubusercontent.com/7364732/192233524-92416a81-624b-487d-ae21-c0f2b97cd6af.png)
![grafik](https://user-images.githubusercontent.com/7364732/192233597-edc50bb0-4fc3-4d04-ac21-d29e73000537.png)

Logs from docker appwrite container:
![grafik](https://user-images.githubusercontent.com/7364732/192233981-16a006a7-4df2-4209-82f4-416b784d87d8.png)

Database:
![grafik](https://user-images.githubusercontent.com/7364732/192234164-128049bd-99ed-44d9-b14a-17115263fd44.png)
![grafik](https://user-images.githubusercontent.com/7364732/192234209-ca256546-58ee-46ec-9b87-5bcc3906ba58.png)
![grafik](https://user-images.githubusercontent.com/7364732/192234226-54c152c9-44c3-4258-9ff5-66c41278ed3a.png)
![grafik](https://user-images.githubusercontent.com/7364732/192234833-aaf25635-b0d5-4398-936d-d30d54ad6cd7.png)
![grafik](https://user-images.githubusercontent.com/7364732/192234865-addc03e8-d630-402c-ac64-ebb089e5c8f4.png)
![grafik](https://user-images.githubusercontent.com/7364732/192234892-df6395d6-9297-4b46-a2e8-d5c8bfa9a3b8.png)

### 👍 Expected behavior

> {
  method: 'GET',
  url: 'https://appwrite.domain.de/v1/databases/62fd0b1b29e929838685/collections/6324f4765e0ff0c5caa6/documents',
  params: { queries: [ 'equal("id", [0])' ] },
  headers: {
    'content-type': 'application/json',
    'x-sdk-name': 'Node.js',
    'x-sdk-platform': 'server',
    'x-sdk-language': 'nodejs',
    'x-sdk-version': '8.0.0',
    'X-Appwrite-Response-Format': '1.0.0',
    'x-appwrite-project': '62fcbdb8436978bcfee3',
    'x-appwrite-key': ''
  },
  data: null,
  json: true,
  responseType: 'json'
}
{
  total: 1,
  documents: [
    {
      id: 0,
      name: 'test',
      '$id': '6324f798d1d2dd4f6f3d',
      '$createdAt': '2022-09-16T22:24:24.859+00:00',
      '$updatedAt': '2022-09-22T12:42:30.304+00:00',
      '$permissions': [],
      '$collectionId': '6324f4765e0ff0c5caa6',
      '$databaseId': '62fd0b1b29e929838685'
    }
  ]
}

### 👎 Actual Behavior

> {
  method: 'GET',
  url: 'https://appwrite.domain.de/v1/databases/62fd0b1b29e929838685/collections/6324f4765e0ff0c5caa6/documents',
  params: { queries: [ 'equal("id", [0])' ] },
  headers: {
    'content-type': 'application/json',
    'x-sdk-name': 'Node.js',
    'x-sdk-platform': 'server',
    'x-sdk-language': 'nodejs',
    'x-sdk-version': '8.0.0',
    'X-Appwrite-Response-Format': '1.0.0',
    'x-appwrite-project': '62fcbdb8436978bcfee3',
    'x-appwrite-key': ''
  },
  data: null,
  json: true,
  responseType: 'json'
}
C:\Users\zwe1ch\Desktop\games\functions\igdb-platforms\node_modules\node-appwrite\lib\client.js:172
                        throw new AppwriteException(error.response.data.message, error.response.status, error.response.data.type, error.response.data);
                              ^
        type: '->',
        args: [ {}, {} ]
      },
      {
        file: '/usr/src/code/app/http.php',
        line: 264,
        function: 'run',
        class: 'Utopia\\App',
        type: '->',
        args: [ {}, {} ]
      }
    ],
    version: '1.0.1'
  }
}

### 🎲 Appwrite version

Version 1.0.x

### 💻 Operating system

Linux

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
Thanks for raising this issue! 🙏 We'll try to get this fixed soon!

**Comment by ElectraPort** [comment]
Team, I know you're a startup and trying to move fast but please take some time to stop adding features and just fix your documentation? I've wasted hours now on bad documentation for "working" product.

In this case, your docs at [your documentation site](https://appwrite.io/docs/databases#querying-documents) say to use a pattern such as:

```
[
    Query.equal('title', ['Avatar', 'Lord of the Rings']),
    Query.greaterThan('year', 1999)
]
```

but to get it working in server-side Node.js (ES6 module style) I had to use:

```
import { Query, InputFile, ID } from 'node-appwrite';
...

[
    `title.equal( ['Avatar', 'Lord of the Rings'] )`
]
```

So please help me understand, as I am trying to write an application using yours, is the appropriate, FINAL pattern the former or the latter?

Do I even need to import Query at all, since I am passing a plain string in a plan array, and not the output of a function call? Signs point to no.

Best,

**Comment by stnguyen90** [comment]
@ElectraPort, for Appwrite version 1.0.x, the documentation is correct. If your second example worked, you may have been using the incorrect version of the SDK with your version of Appwrite.

If you're still having troubles, feel free to reach out on [Discord](https://appwrite.io/discord).

**Comment by ElectraPort** [comment]
For the future reader, while there was an outage in documentation and it cost me some hours in searching between this and an issue with Storage and the InputFile object, the following are true:
- We signed up for a "beta" by installing a v0.x product;
- We ignored the suggestion to upgrade to v1.0x when it came out;
- Look at the elapsed time between my original post and now. I already upgraded our platform from v.0.15.3.402 where the pattern in my post works (a string) to the current version (1.0.1.500), where the pattern in the AppWrite documentation works as documented. 

Stay current, my friends. 

Thanks for your help @stnguyen90 !
